<launch>
  <arg name="launch_prefix" default="" /> <!-- set to value="gdbserver localhost:10000" for remote debugging -->
  <arg name="node_namespace" default="camera" />
  <arg name="camera_name" default="calibration_target" />
  <arg name="camera_info_dir" default="$(find tag_utils)/camera_info" />
  <arg name='use_webcam' default="false" />
  <arg name="capture_width" default="640" />
  <arg name="capture_height" default="480" />
  <arg name="image_topic" default="image_rect" />
  
  <node pkg="tag_utils" type="camera_node" name="$(arg node_namespace)" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)">
    <!-- Remap topics from those used in code to those on the ROS network -->
    <param name="camera_name" value="$(arg camera_name)" />
    <param name="camera_info_url" value="$(arg camera_info_dir)/$(arg camera_name).yaml" />
    <param name="use_webcam" value="$(arg use_webcam)" />
    <param name="capture_width" value="$(arg capture_width)" />
    <param name="capture_height" value="$(arg capture_height)" />
    
    <remap from="$(arg camera_name)/image_rect" to="camera/image_rect" />
  </node>

  <arg name="square" default="0.2" />
  <arg name="size" default="8x6" />
 <node pkg="camera_calibration" type="cameracalibrator.py" name="calibrator" args="--size $(arg size) --square $(arg square) image:=/camera/image_rect camera:=/camera" >
  </node>
  
</launch>
